### ISSUE TITLE
Low performance on several datasets

### ISSUE THREAD

**Issue by hsfzxjy** [shipra25jain/ESSNet#5]
Hi. I've tried training the models on several datasets with recommended settings, but all of them achieved relatively low performance. The results are as below:

| model | dataset | d|k|bs| lr | iters | mIoU |
| ------ | ---------| ---|--|--| --| ----|--------|
|DeepLabV3_mobilenet| Cityscapes | 6|7|12|0.01|50k|66.2|
|DeepLabV3_mobilenet| Cityscapes | 6|7|12|0.001|50k|~50|
|DeepLabV3_resnet50| ade20k|12|8|10|0.01|85k|34.4|
|DeepLabV3_resnet50| cocostuff-10k|12|7|10|0.01|85k|29.7|

The logs are attached here:
[cityscapes.txt](https://github.com/shipra25jain/ESSNet/files/6202810/cityscapes.txt)
[cocostuff.txt](https://github.com/shipra25jain/ESSNet/files/6202811/cocostuff.txt)
[ade20k.txt](https://github.com/shipra25jain/ESSNet/files/6202812/ade20k.txt)

I've made some minor modification to the code, but I think they are not concerned with the results. The changes can be found in my forked repo: https://github.com/hsfzxjy/ESSNet/

Thanks and looking forward to your advice!

**Comment by shipra25jain** [comment]
Hello @hsfzxjy ,

As far as I remember, we trained cityscapes with higher learning rate may be 0.1 and also number of iterations were higher than what you used for ade20k and cocostuff. If needed, I can dig checkpoints of our experiments and share with you. Though I am not sure, if I have all the checkpoints store, may be the last checkpoint of training should be somewhere. Let me know if that would help.

**Comment by hsfzxjy** [comment]
@shipra25jain Thanks. It will be helpful if the checkpoints/logs be shared.

Besides, the training iterations are followed by your Appendix A
![image](https://user-images.githubusercontent.com/4702188/112578031-7eb07500-8e30-11eb-9b31-39130f95e680.png)
For Cityscapes, the total iterations should be: 200 (epochs) * 2975 (images) / 12 (batch size) = ~50k

And also in Section 4.1, the paper mentioned that base lr for Cityscapes should be 1e-3
![image](https://user-images.githubusercontent.com/4702188/112580867-478e9380-8e31-11eb-81b3-6f3379cbb0b5.png)

Looking forward to your reply!

**Comment by shipra25jain** [comment]
Hello @hsfzxjy ,
I checked the logs and tensorboard summaries, and I see that we have used 0.1 lr for cityscapes experiment and also set max iterations to 100k (learning rate curve also depends upon this). However, our model converged somewhere around 50-60k iterations, so we did early stopping. In my laptop I only have the last checkpoint of models and do not have access to the machine which has all checkpoints stored. I am attaching the checkpoints of cityscapes with MobileNet  (d=7)and ResNet50 (they might not have exact same scores as in paper as this is the checkpoint where I stopped training).  Also, I am attaching ade20k ResNet50 checkpoint (d=12). For coco-stuff, I do not have the backup of checkpoints. I would recommend you to train with higher number of max iterations for bigger datasets like 150k for ade20k and coco stuff. Though model will achieve best scores much earlier, the higher max iteration would ensure proper learning rate schedule.
You will find checkpoints here : https://drive.google.com/drive/folders/13nErSx92L0r1Vp6yvKfUyoIREOMLB-cj?usp=sharing

I will make correction in paper for cityscape lr. 
Thank you
Shipra

**Comment by hsfzxjy** [comment]
@shipra25jain Thanks for your reply. I will try the recommended learning rate schedule.

**Comment by hsfzxjy** [comment]
@shipra25jain Hi, I've reproduced with the new LR settings as follow:

1. On Cityscapes, training with LR=0.1, 150k iters but stop at 100k iters, MobileNetv2 achieves 73.5 mIoU (comparable with 73.03);
2. On ADE20K, training with LR=0.01, 150k iters, ResNet50 achieves 38.11 mIoU (slightly lower than 38.29);
3. On COCOStuff-10k, training with LR=0.01, 150k iters, ResNet50 achieves 31.36 mIoU (lower than 32.60).

I think the results are now acceptable. Thanks!
